tests/type-checks/mypy_typing_of_has_logger.py

```diff
@@ -9,3 +15,32 @@ def verify_has_logger_type() -> logging.Logger:
 
 def verify_has_extended_debug_logger_type() -> ExtendedDebugLogger:
     return HasExtendedDebugLogger.logger
+
+
+GenericWithLoggerMeta = HasLoggerMeta.meta_compat(GenericMeta)
+
+
+# mypy doesn't recognize this as a valid metaclass but it still functions
+# properly
+class GenericWithLogger(Generic, metaclass=GenericWithLoggerMeta):  # type: ignore
+    pass
+
+
+T = TypeVar("T")
+
+
+# mypy gets confused about the actual provision of the types for the generics
+# but the actual type checking mechanism below works.
+class VerifyGenericWithLoggerTyping(GenericWithLogger[T]):  # type: ignore
+    @classmethod
+    def return_T(cls, value: T) -> T:
+        return value
+
+
+class VerifyGenericWithLoggerTypingInteger(VerifyGenericWithLoggerTyping[int]):
+    pass
+
+
+int_value: int
+
+int_value = VerifyGenericWithLoggerTypingInteger.return_T(1234)
```

Comment:
Hm, any way we can test a failing type check? Would be nice to have a strict `xfail` for type checking.

Seems like this test would pass if all the `Generic` machinery stopped working, or the mypy test wasn't running.

Reply:
The best I could come up with was using a `type: ignore` comment which I think is better than nothing since an `Any` type slipping through or something would test as a false positive without that.

I also did manual testing to ensure that all the types are as expected which doesn't give us long-term guarantees but it does give me some confidence that this is indeed actually working as expected.